### PR TITLE
feat(linter): Implement `eslint/prefer-object-has-own`

### PR DESCRIPTION
**Pull request by tbashiyy**
I added eslint rule `eslint/prefer-object-has-own`.
https://eslint.org/docs/latest/rules/prefer-object-has-own

This is my first PR for this repository🙏

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+        "class C { #hasOwnProperty; foo() { ({}.#hasOwnProperty.call(obj, prop)) } }",
+        "class C { #call; foo() { ({}.hasOwnProperty.#call(obj, prop)) } }",
+        "({ foo }.hasOwnProperty.call(obj, prop))",
+        "(Object) => ({}).hasOwnProperty.call(obj, prop)",
+        r#"
+			        let obj = {};
+			        Object.hasOwn(obj,"");
+			        "#,
+        "const hasProperty = Object.hasOwn(object, property);",
+        // "/* global Object: off */
```
Comment (tbashiyy) [inline crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs:213]: I have a question.
I don't know how do I handle this comment in oxc ast.
Currently, I commented out.
If I have to handle this test case, please tell me.

File: crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+    conditional_fix
+);
+
+impl Rule for PreferObjectHasOwn {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+
+        let Some(callee) = unwrap_parenthesized_expr(&call_expr.callee).get_member_expr() else {
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs:67]: I think you want `get_inner_expression`

File: crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+        let Some(object) = unwrap_parenthesized_expr(callee.object()).get_member_expr() else {
+            return;
+        };
+
+        let callee_property_name = callee.static_property_name();
+        let object_property_name = object.static_property_name();
+        let is_object = has_left_hand_object(object);
+        let is_global_scope = ctx.scopes().find_binding(node.scope_id(), "Object").is_none();
+
+        if callee_property_name == Some("call")
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs:80]: I think you want `is_method_call`

File: crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs (hunk 1 above)
Comment (DonIsaac) [reply crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs:213]: My approach so far has been "using comments to declare globals is stupid and outdated". Feel free to ignore such cases.

**Review by DonIsaac — COMMENTED** [review]
Thanks for the PR! Welcome to Oxc :)

File: crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs (hunk 2 above)
Comment (tbashiyy) [reply crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs]: thanks!
fixed af29d35e83c269a695965c17f4e4063d265168ff

File: crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs (hunk 3 above)
Comment (tbashiyy) [reply crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs]: thanks!
fixed 90ac4054de6ddaa4db178e0b61ad4e2cda2520d1

**Review by DonIsaac — APPROVED** [review]

**Pull request merged**